golem_messages/message/tasks.py

```diff
@@ -388,7 +375,10 @@ def generate_ethsig(
                 value=self.requestor_ethereum_public_key,
             )
 
-        self._ethsig = self._get_signature(private_key, msg_hash)
+        if msg_hash is None and self.want_to_compute_task is not None:
```

Comment:
why? .... I believe:

`self.ethsig = self.want_to_compute_task._get_signature(private_key, msg_hash)` should do the trick...

Reply:
The question is if there should be still possibility to create `ethsig` with custom `msg_hash` passed to this function. As this possibility was there before, I have assumed it should still be there for optimization purposes, as in the method comment `to skip generation of the message hash while signing`. Should I remove it then?

Reply:
Not exactly, if `msg_hash` will be `None,` then `_get_signature` will try to create it from `self.get_short_hash()`, when `self` is `TaskToCompute` in this context.

Reply:
well, not really, because `_get_signature()` will be called from `WantToComputeTask` (in `want_to_compute_task`) and not from `self.` ...

Reply:
:+1: done